fluentd/generate_throttle_configs.rb

```diff
@@ -38,24 +47,75 @@ def get_file_name(name)
   return file_name
 end
 
+## Returns the names of all throttle configs for parsing through when reverting
+def get_all_throttle_files()
+  return Dir.glob('/var/log/es-container-*.log.pos')
+end
+
+def move_pos_file_project_entry(source_file, dest_file, project)
+
+  if File.file?(source_file)
+    project_pattern = ".*_#{project}_.*\.log"
+
+    matches = Array.new()
+    File.open(source_file) { |file|
+      file.grep(/#{project_pattern}/) { |match| matches << match }
+    }
+
+    update = ""
+    update = File.read(dest_file) if File.file?(dest_file)
+    matches.each { |match|
+      log_file_pattern = Regexp.escape(match.split(" ")[0])
+      update.gsub!(/#{log_file_pattern}.*$\n/, "")
+      ## We are clearing out the line while if it exists and then appending the source_file line
+      update << match
+    }
+    File.open(dest_file, "w") { |file| file.puts update }
```

Comment:
Should this be 'write' instead of 'puts'... might make it more obvious what is happening

Reply:
we can probably use `write` over `puts` since we don't necessarily need an extra carriage return at the end of `update`...